### PR TITLE
Remove the need for cache.Parent

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by deltakosh** [comment]
Setting as draft to avoid unwanted merge during code freeze

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14888/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14888/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14888/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14888/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14888/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14888/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14888/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14888/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14888/merge/testResults/webgl2playwright/index.html

**Review by sebavan — APPROVED** [review]

**Comment by RaananW** [comment]
Unit test fail - Mesh Baking › bakeCurrentTransformIntoVertices › should set children transforms from their parent

**Comment by deltakosh** [comment]
> Unit test fail - Mesh Baking › bakeCurrentTransformIntoVertices › should set children transforms from their parent

It was a wrong test all along :) Happy to have catched that issue

File: packages/dev/core/src/node.ts [hunk 1]
```diff
@@ -239,6 +239,7 @@ export class Node implements IBehaviorAware<Node> {
 
         // Store new parent
         this._parentNode = parent;
+        this._isDirty = true;
```
Comment (gemini133) [inline packages/dev/core/src/node.ts:242]: why add this? it just breaks some of my models

**Pull request merged**